### ISSUE TITLE
splitFile option doesn't support a resource from a jar

### ISSUE THREAD

**Issue by meriouma** [splitio/java-client#390]
This works when the resource is loaded from the filesystem when running the app through the IDE, but it doesn't when the resource is loaded from a jar. I believe `splitFile` should support an URI to call `new File(...)` with a `jar:` URI.
```
val splitFile = javaClass.classLoader.getResource("split.yml")
val splitClientConfig = SplitClientConfig.builder()
        .splitFile(splitFile?.file)
        .build()
```

Something like this :
```
val splitFile = javaClass.classLoader.getResource("split.yml")
val splitClientConfig = SplitClientConfig.builder()
        .splitFile(splitFile?.toURI())
        .build()
```

**Comment by agustinona** [comment]
Hi @meriouma I can submit this as a feature request on your behalf for our team to evaluate. Could you provide a use case? What's the advantage of this approach over providing the path to the split.yml file to the splitFile method?

**Comment by meriouma** [comment]
Our split.yml file is in the resource folder of our app. We are compiling a native application with GraalVM that we put in a Docker image, so our expectation is that we can load the split.yml that was bundled with the app. We don't want to copy the file to the Docker image manually, as the file is already in the jar, and the Dockerfile is generated by another tool.

**Comment by agustinona** [comment]
@meriouma Passing the relative path to the file should work for the bundled yml in both the IDE and when running the jar:

`.splitFile("foo/bar/resources/split.yml")`

Is this not a workable solution for your case?

**Comment by agustinona** [comment]
I believe this issue stems from your usage of `javaClass.classLoader.getResource` to get the filePath, since it returns an absolute path which can be problematic after bundling into a jar file. If you instead provide a relative path (i.e. a path which doesn't start with "/") it will use that path to find the file starting from the project's root. This will work the same within or without the jar file.
 
For example, if I have a `featureflags.yml` file with my feature flag definitions at my project's root I can build my factory as follows:
```
config = SplitClientConfig.builder()
.splitFile("featureflags.yml")
.build();
this.myFactory = SplitFactoryBuilder.build("localhost", config);
```
This approach will work the same after the app is bundled into a jar file.

**Comment by agustinona** [comment]
@meriouma I'd like to move this into a support ticket for better tracking and more agile troubleshooting.
You should have received an email from support@split.io. Please let me know if you haven't.

**Comment by mchwedczuk-box-com** [comment]
+1 I have exactly the same problem. In our current setup we run integration tests using a fat jar created by Spring Boot.

I think instead of relying on specific Path's or URLs there should be an option to pass the config as `InputStream` that would allow people to use any implementation they want to provide the default configuration.

**Comment by nmayorsplit** [comment]
Hi @meriouma and @mchwedczuk-box-com, 
Sorry for the delay on this. We've released JAVA `4.9.0` which supports passing "local file" as InputStream.
Below is an example of how to use it:

InputStream inputStream = new FileInputStream("parentRoot/split.yaml");
SplitClientConfig config = SplitClientConfig.builder()
       .splitFile(inputStream, FileTypeEnum.YAML)
       .setBlockUntilReadyTimeout(10000)
       .build();
SplitClient client = SplitFactoryBuilder.build("localhost", config).client();

Here is the link to our official docs [https://help.split.io/hc/en-us/articles/360020405151-Java-SDK#localhost-mode]

Thanks,
Nadia
